analytics/src/main/java/com/segment/analytics/Analytics.java

```diff
@@ -1310,6 +1315,16 @@ public Builder useDestinationMiddleware(String key, Middleware middleware) {
       return this;
     }
 
+    public Builder useEdgeFunctionMiddleware(JSMiddleware middleware) {
+
```

Comment:
We should add an IllegalStateException here, in the case that sourceMiddleware/destinationMiddleware are already used